### PR TITLE
Implement missions joining and leaving: Actions

### LINKED ISSUE #9: [1pt] Implement mission leaving - Actions
- Follow the same logic as with the "Join mission" - but you need to set the `reserved` key to `false`. 
- Dispatch these actions upon click on the corresponding buttons.

### LINKED ISSUE #10: [3pt] Implement mission joining - Actions
- When a user clicks the "Join Mission" button, action needs to be dispatched to update the store. You need to get the ID of the selected mission and update the state. Remember you mustn't mutate the state. Instead, you need to return a new state object with all missions, but the selected mission will have an extra key `reserved` with its value set to `true`. You could use a JS `filter()` or `map()` to set the value of the new state - i.e.:
```javascript
const newState = state.map(rocket => {
    if(mission.id !== id) 
        return mission;
    return { ...mission, reserved: true };
});

```

- Regardless of which method you choose, make sure you place all your logic in the reducer. In the React view file, you should only dispatch the action with the correct rocket ID as an argument.

### PR DESCRIPTION
**Pull request by jerryowusu**
## Description

In this milestone, I created actions for joining and leaving missions.

### General requirements

- [x] There are **no linter errors**.
- [x] Followed the **GitFlow**.
- [x] The documentation is **professional**.

### HTML/CSS requirements

- [x] Followed best practices for JavaScript

### Project requirements

- [x] When a user clicks the "Join Mission" button, action needs to be dispatched to update the store. You need to get the ID of the selected mission and update the state.
- [x] Regardless of which method you choose, make sure you place all your logic in the reducer. In the React view file, you should only dispatch the action with the correct rocket ID as an argument.
-[x] Dispatch these actions upon click on the corresponding buttons.

### Additional notes

closes #10 
closes #9 

---
Thank you for taking the time to review this pull request.

### CODE REVIEW

**Review by mirouhml — APPROVED** [review]
Hi @jerryowusu well done!

Your project is complete! I have a small suggestion to ask for but other than that it's time to merge it. Congratulations! 🎉


Cheers and Happy coding!👏👏👏

File: src/components/Mission.js [hunk 1]
```diff
@@ -1,13 +1,41 @@
...
+      <td><p className="member">NOT A MEMBER</p></td>
+      <td>
+        <button
+          className="btn"
+          onClick={handleSubmit}
+          type="button"
+        >
+          {joined ? 'Leave Mission' : 'Join Mission'}
+        </button>
+      </td>
```
Comment (mirouhml) [inline src/components/Mission.js:38]: The "Leave Mission" button style changes back to black (default style) when you switch to another page and then go back to the missions page.
image
I suggest you look into that, maybe add another condition for the style itself.

Comment (jerryowusu) [reply src/components/Mission.js:38]: Thanks for alerting me on this. I will try and resolve that.

**Comment by jerryowusu** [comment]
Thanks

**Pull request merged**